entrypoint.sh

```diff
@@ -4,12 +4,23 @@ image=$1
 report=$2
 patched_tag=$3
 timeout=$4
+output_file=$5
+format=$6
+
 
 # parse image into image name
 image_no_tag=$(echo "$image" | cut -d':' -f1)
 
+# check if output_file has been set
+if [ -z "$output_file" ]
+then
+    output=""
+else
+    output="--format $format --output ./data/"$output_file""
+fi
+
 # run copa to patch image
-if copa patch -i "$image" -r ./data/"$report" -t "$patched_tag" --addr tcp://127.0.0.1:8888 --timeout $timeout;
+if copa patch -i "$image" -r ./data/"$report" -t "$patched_tag" --addr tcp://127.0.0.1:8888 --timeout $timeout $output;
```

Comment:
output json won't have user read permissions. do we need to set up permissions? how does other actions handle this case?

Reply:
@Vombato take a look at [this](https://docs.github.com/en/actions/creating-actions/creating-a-docker-container-action#accessing-files-created-by-a-container-action)

Reply:
> since this is created from inside the container, output json won't have user read permissions. do we need to set up permissions? how does other actions handle this case?

This should be true only for the "build" action that produces the copa-action but not for the copa-action itself, since here: https://github.com/Vombato/copa-action/blob/45ad120c0502d23e649ab235a843ea2f24097602/action.yaml#L53C1-L53C434
the workspace path is mounted in the /data directory which should also be the directory in which copa writes the output file.
Am I missing something? Do we have means to test this?

Reply:
You can run the action against your branch. Here's my test: https://github.com/sozercan/copa-action/actions/runs/7674100022/job/20918050594#step:6:11

`-rw------- 1 root   root    49735 Jan 26 22:28 test.json`

Reply:
@sozercan @vombato
The GitHub action for Trivy has an [open issue] (https://github.com/aquasecurity/trivy-action/issues/283) from two months ago, addressing the exact same issue.  
Examining another [related issue] (https://github.com/actions/runner/issues/1282) and its linked issues reveals that the problem arises because the user is not specified in the action's Dockerfile.  However, specifying the user results in the [loss of several functionalities] (https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#docker-container-filesystem).  
At this point, the simplest course of action seems to be adding an instruction to change the file permissions directly in our entrypoint script.

Reply:
@R3DRUN3 good find! can we set it to `444` like trivy action does?

Reply:
> @R3DRUN3 good find! can we set it to `444` like trivy action does?

I added a chmod 444 that should solve the issue, tests were ok, please check for yourself and give us feedback if needed 😃